pkgs/development/libraries/kde-frameworks/kwindowsystem/default.nix

```diff
@@ -10,11 +10,5 @@ mkDerivation {
   nativeBuildInputs = [ extra-cmake-modules ];
   buildInputs = [ libpthreadstubs libXdmcp qttools qtx11extras ];
   propagatedBuildInputs = [ qtbase ];
-  patches = [
-    ./0001-platform-plugins-path.patch
```

Comment:
We need to load `kwayland-integration` as `kwindowmanager` plugin, but `kwayland-integration` depends on `kwindowmanager`

I see no way here to make it pure

Reply:
Might be possible by finding out what parts of `kwindowmanager` it depends on exactly and splitting that into a separate output. Or maybe it would be possible to set something up with env variables in the NixOS module?

Reply:
> Might be possible by finding out what parts of `kwindowmanager` it depends on exactly and splitting that into a separate output

It is pretty much monolithic

> Or maybe it would be possible to set something up with env variables in the NixOS module?

Well, it is possible to introduce new env variable in plasma5 nixos module, but this doesn't solve impurity, as environment needs to be propagated :D